File: frontend/src/app/services/terminal-vendor/terminal-vendor.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { TerminalVendor } from '../../models/TerminalVendor';
import { ResponseEntityData } from '../../models/ResponseEntityData';

@Injectable({
  providedIn: 'root'
})
export class TerminalVendorService {
  private readonly baseURL = 'http://localhost:8081/api/terminal-vendors';

  constructor(private readonly http: HttpClient) {}

  createTerminalVendor(terminalVendor: TerminalVendor): Observable<ResponseEntityData> {
    return this.http.post<ResponseEntityData>(this.baseURL, terminalVendor);
  }

  getAllTerminalVendors(): Observable<ResponseEntityData> {
    return this.http.get<ResponseEntityData>(this.baseURL);
  }

  getAllTerminalVendorsIdAndName(): Observable<ResponseEntityData> {
    return this.http.get<ResponseEntityData>(`${this.baseURL}/filter`);
  }

  getTerminalVendorById(id: number): Observable<ResponseEntityData> {
    return this.http.get<ResponseEntityData>(`${this.baseURL}/${id}`);
  }

  updateTerminalVendor(id: number, terminalVendor: TerminalVendor): Observable<ResponseEntityData> {
    return this.http.put<ResponseEntityData>(`${this.baseURL}/${id}`, terminalVendor);
  }

  deleteTerminalVendor(id: number): Observable<ResponseEntityData> {
    return this.http.delete<ResponseEntityData>(`${this.baseURL}/${id}`);
  }
}

File: backend/src/main/java/com/bnt/rcp/service/TerminalVendorService.java
package com.bnt.rcp.service;

import java.util.List;

import com.bnt.rcp.dto.TerminalVendorDto;

public interface TerminalVendorService {
    TerminalVendorDto createTerminalVendor(TerminalVendorDto terminalVendorDto);
    TerminalVendorDto getTerminalVendorById(Integer id);
    List<TerminalVendorDto> getAllTerminalVendors();
    TerminalVendorDto updateTerminalVendor(Integer id, TerminalVendorDto terminalVendorDto);
    void deleteTerminalVendor(Integer id);
}
